grpc_unary_attestation/src/client.rs

```diff
@@ -16,98 +16,70 @@
-    AttestationBehavior, ClientHandshaker, Encryptor, ServerIdentityVerifier,
+use async_trait::async_trait;
+use oak_remote_attestation::handshaker::ServerIdentityVerifier;
+use oak_remote_attestation_sessions::{
+    client::{GenericAttestationClient, UnaryClient},
+    SessionId,
 };
-use oak_remote_attestation_sessions::SessionId;
 use tonic::transport::Channel;
 
-/// gRPC Attestation Service client implementation.
-pub struct AttestationClient {
-    session_id: SessionId,
-    encryptor: Encryptor,
-    client: UnarySessionClient<Channel>,
+pub struct GrpcClient {
+    inner: UnarySessionClient<Channel>,
 }
 
-impl AttestationClient {
-    pub async fn create(
-        uri: &str,
-        expected_tee_measurement: &[u8],
-        server_verifier: ServerIdentityVerifier,
-    ) -> anyhow::Result<Self> {
-        let session_id: SessionId = rand::random();
+impl GrpcClient {
+    pub async fn create(uri: &str) -> anyhow::Result<Self> {
         let channel = Channel::from_shared(uri.to_string())
             .context("Couldn't create gRPC channel")?
             .connect()
             .await?;
-        let mut client = UnarySessionClient::new(channel);
-
-        let mut handshaker = ClientHandshaker::new(
-            AttestationBehavior::create_peer_attestation(expected_tee_measurement),
-            server_verifier,
-        );
-        let client_hello = handshaker
-            .create_client_hello()
-            .context("Couldn't create client hello message")?;
+        let inner = UnarySessionClient::new(channel);
+        Ok(Self { inner })
+    }
+}
 
-        let mut response = client
+#[async_trait]
+impl UnaryClient for GrpcClient {
+    async fn message(&mut self, session_id: SessionId, body: Vec<u8>) -> anyhow::Result<Vec<u8>> {
+        let response = self
+            .inner
             .message(UnaryRequest {
-                body: client_hello,
+                body,
                 session_id: session_id.to_vec(),
             })
             .await
-            .context("Couldn't send client hello message")?
-            .into_inner();
+            .context("Couldn't send message")?
+            .into_inner()
+            .body;
 
-        while !handshaker.is_completed() {
-            let request = handshaker
-                .next_step(&response.body)
-                .context("Couldn't process handshake message")?;
+        Ok(response)
+    }
+}
 
-            if let Some(request) = request {
-                response = client
-                    .message(UnaryRequest {
-                        body: request,
-                        session_id: session_id.to_vec(),
-                    })
-                    .await
-                    .context("Couldn't send client hello message")?
-                    .into_inner();
-            }
-        }
+/// gRPC Attestation Service client implementation.
+pub struct AttestationClient {
+    inner: GenericAttestationClient<GrpcClient>,
+}
```

Comment:
This feels like it should be in a separate crate (see my other comment too).

Reply:
Agreed about splitting out the std client from the rest of `remote_attestation_sessions`. But why should this be a separate crate? This piece of code is just the existing interface of the gRPC client, now changed to use the `GenericAttestationClient` under the hood. 

The diff is somewhat misleading by showing this as new code, which for the most part it isn't. :)

Reply:
Diff aside (I'm not trying to figure out whether it's pre-existing or not), my point was whether we should try and keep client and server libraries separate. But maybe we don't care that much about the gRPC specific logic, since even in its server side component it's not expected to be part of the bare-metal binary.

Reply:
Oh, I see, the suggestion was about splitting this crate into two as well. :) 

Imo for the gRPC boys one crate is fine. Like you said, it wouldn’t be part of bare metal either way